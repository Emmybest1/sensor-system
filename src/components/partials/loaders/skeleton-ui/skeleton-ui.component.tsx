import React from 'react';
import styled, { keyframes } from 'styled-components';

const skeletonUiDimmer = keyframes`
 from {
      background-color: #e2e2e2;
    }
  
    50% {
      background-color: #f3f3f3;
    }
  
    to {
      background-color: #e2e2e2;
    }
 `;

type SkeletonUiPropTypes = {
  shape: 'radiusHorizone' | 'radiusVerticone' | 'blockHorizone' | 'blockVerticone';
  width?: string;
  height?: string;
};

const SkeletonUiNode = styled.div<SkeletonUiPropTypes>`
  animation: ${skeletonUiDimmer} 1.5s infinite;
  margin-bottom: 0.625rem;
  border-radius: ${({ shape }) => (['radiusHorizone', 'radiusVerticone'].includes(shape) ? '0.1875rem' : '0')};
  width: ${({ shape, width }) => width || (['radiusVerticone', 'blockVerticone'].includes(shape) ? '1.25rem' : '100%')};
  height: ${({ shape, height }) => height || (['radiusHorizone', 'blockHorizone'].includes(shape) ? '1.25rem' : '100%')};
`;

export const SkeletonUi: React.FC<SkeletonUiPropTypes> = (props) => <SkeletonUiNode {...props} />;
