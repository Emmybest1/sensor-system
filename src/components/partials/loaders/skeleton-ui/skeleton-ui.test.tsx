import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { SkeletonUi } from './skeleton-ui.component';

describe('<SkeletonUi/>', () => {
  test('should render Component', () => {
    const { container } = render(
      <Router>
        <SkeletonUi shape="blockHorizone" />
        <SkeletonUi shape="blockVerticone" />
        <SkeletonUi shape="radiusHorizone" width="31.25rem" height="2rem" />
        <SkeletonUi shape="radiusVerticone" width="90%" height="2rem" />
      </Router>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
