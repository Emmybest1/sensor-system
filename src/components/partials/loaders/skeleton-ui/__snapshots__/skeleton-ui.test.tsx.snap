// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkeletonUi/> should render Component 1`] = `
<div
  class="sc-bdfBQB bKdmMW"
  shape="blockHorizone"
/>
`;
